### PR TITLE
Add PodDisruptionBudget syncer

### PR DESCRIPTION
**Pull request by pratikjagrut**
This is PR is regarding issue  #72. It adds syncer for the `PodDisruptionBudget`.

### CODE REVIEW

**Review by matskiv — APPROVED** [review]
Code looks great :+1:

File: charts/k0s/templates/rbac/role.yaml [hunk 1]
```diff
@@ -22,6 +22,9 @@ rules:
   - apiGroups: ["apps"]
     resources: ["statefulsets", "replicasets", "deployments"]
     verbs: ["get", "list", "watch"]
+  - apiGroups: ["policy"]
```
Comment (FabianKramm) [inline charts/k0s/templates/rbac/role.yaml:25]: Should be moved to .Values.rbac.role.extended below

File: charts/k3s/templates/rbac/role.yaml [hunk 2]
```diff
@@ -22,6 +22,9 @@ rules:
   - apiGroups: ["apps"]
     resources: ["statefulsets", "replicasets", "deployments"]
     verbs: ["get", "list", "watch"]
+  - apiGroups: ["policy"]
```
Comment (FabianKramm) [inline charts/k3s/templates/rbac/role.yaml:25]: Should be moved to .Values.rbac.role.extended below

File: charts/k8s/templates/rbac/role.yaml [hunk 3]
```diff
@@ -22,6 +22,9 @@ rules:
   - apiGroups: ["apps"]
     resources: ["statefulsets", "replicasets", "deployments"]
     verbs: ["get", "list", "watch"]
+  - apiGroups: ["policy"]
```
Comment (FabianKramm) [inline charts/k8s/templates/rbac/role.yaml:25]: Should be moved to .Values.rbac.role.extended below

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@pratikjagrut thanks a lot for the PR! Looks good to me, only thing is that we should move rbac permissions to the role.extended area.

File: charts/k0s/templates/rbac/role.yaml (hunk 1 above)
Comment (pratikjagrut) [reply charts/k0s/templates/rbac/role.yaml]: done

File: charts/k8s/templates/rbac/role.yaml (hunk 3 above)
Comment (pratikjagrut) [reply charts/k8s/templates/rbac/role.yaml]: done

File: charts/k3s/templates/rbac/role.yaml (hunk 2 above)
Comment (pratikjagrut) [reply charts/k3s/templates/rbac/role.yaml]: done

**Review by FabianKramm — APPROVED** [review]
@pratikjagrut thanks! LGTM!

**Pull request merged**